guardian/cdk
# fix: Switch from our home baked SSM policy to the current managed policy

## What does this change?

This PR effectively adds some extra SSM permissions required for AWS Inspector V2.

Once upon a time the recommended managed policy allowed full access to all S3 buckets but this craziness [is no longer the case and the managed policy is treated with care](https://aws.amazon.com/blogs/mt/applying-managed-instance-policy-best-practices/). We should switch to this, at least in part, because it has some extra permissions needed for AWS Inspector (v2) and new required SSM permissions will be added for us in the future.

## How to test

This has been and can be tested by doing the following:
 - in the CDK playground project:
   - create a new branch
   - switch to using this CDK branch with `script/switch-cdk sihil/switch-to-managed-ssm-policy`
   - update the test snapshots
   - commit, push and open a draft PR to trigger a build
 - deploy your branch of the CDK playground
 - in to the developer playground account:
   - locate the CDK playground EC2 instance
   - open the IAM role
   - check that the managed role is there and no inline policy relating to SSM exists
 - check you can `ssm ssh` to the EC2 instance with developer playground CLI creds



## How can we measure success?

We'll be able to roll out AWS Inspector v2.

## Have we considered potential risks?

It's possible that this is incorrect and SSM access will break for instances. This would likely break a few things like ES node rotation but is unlikely to cause user facing issues.

## Checklist


[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?
